### PR TITLE
chore: add pnpm instead of yarn

### PR DESCRIPTION
**Pull request by Szymon-dziewonski**
User better and faster package manager

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,26 +12,31 @@ jobs:
...
-      - name: Configure Yarn cache
+      - name: Generate pnpm cache-key
+        id: pnpm-cache-key
+        run: echo "::set-output name=key::${{ runner.os }}-pnpm-ci-cache-${{ hashFiles('./pnpm-lock.yaml') }}"
+      - name: Configure PNPM cache
         uses: actions/cache@v3
         with:
-          key: ${{ steps.yarn-cache-key.outputs.key }}
+          key: ${{ steps.pnpm-cache-key.outputs.key }}
           path: |
```
Comment (FRSgit) [inline .github/workflows/ci.yml:36]: No need for the cache https://github.com/actions/setup-node#caching-global-packages-data

Something like this is sufficient:
```
- name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v2
        with:
          node-version: ${{ matrix.node-version }}
          cache: 'pnpm'
 ```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -12,26 +12,31 @@ jobs:
...
-      - uses: actions/setup-node@v3
+      - name: Install Node.js
+        uses: actions/setup-node@v3
         with:
           node-version: '16.x'
+      - uses: pnpm/action-setup@v2.2.2
+        name: Install pnpm
+        with:
+          version: 7.5.0
+          run_install: false
```
Comment (FRSgit) [inline .github/workflows/ci.yml:26]: run_install is false by default

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -42,19 +47,23 @@ jobs:
...
         uses: actions/cache@v3
         with:
-          key: ${{ needs.cache.outputs.yarn-cache-key }}
+          key: ${{ needs.cache.outputs.pnpm-cache-key }}
           path: |
-            .yarn
             node_modules
       - name: Install dependencies
-        run: yarn --immutable
+        run: pnpm i --frozen-lockfile
```
Comment (FRSgit) [inline .github/workflows/ci.yml:64]: On CI `frozen-lockfile` is `false` by [default](https://pnpm.io/cli/install#--frozen-lockfile)

Comment (Szymon-dziewonski) [reply .github/workflows/ci.yml:64]: you mean `true` right? but sure thanks will remove it

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Szymon-dziewonski) [reply .github/workflows/ci.yml:26]: Not really, its `run_install
Optional (default: null) If specified, run pnpm install.` 
and when I run CI steps without it took much more time, even in their example it shows to have `run_install: false`

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Szymon-dziewonski) [reply .github/workflows/ci.yml:36]: I need to cache it, under your link there is `Note: The action does not cache node_modules`

**Pull request merged**